src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/commands.py

```diff
@@ -0,0 +1,20 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import CliCommandType
+
+from azure.cli.command_modules.natgateway._client_factory import cf_nat_gateways
+
+def load_command_table(self, _):
+
+    nat_gateway_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.network.operations#NatGatewaysOperations.{}',
+        client_factory=cf_nat_gateways)
+
+    with self.command_group('network nat-gateway', nat_gateway_sdk) as g:
+        g.show_command('show', 'get')
+        g.custom_command('update', 'update_nat_gateways')
+        g.custom_command('create', 'create_nat_gateway')
```

Comment:
If create and update are LRO, you should expose the `--no-wait` flag as well as the `wait` command. 
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#register-commands (see item 3 for guidance on this command).